iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/NetworkImpl.java

```diff
@@ -1083,7 +1083,7 @@ private void checkMergeability(NetworkImpl otherNetwork) {
     private static void createSubnetwork(NetworkImpl parent, NetworkImpl original) {
         // The root network reference should point to parent and not original anymore
```

Comment:
Could you just mention like in flatten why you need to have a ref pointing to the same RefChain as the parent network?